flink-runtime/src/test/java/org/apache/flink/runtime/metrics/util/MetricUtilsTest.java

```diff
@@ -79,6 +84,23 @@ public void testNonHeapMetricsCompleteness() {
 		Assert.assertNotNull(nonHeapMetrics.get(MetricNames.MEMORY_MAX));
 	}
 
+	@Test
+	public void testMetaspaceCompleteness() {
+		final InterceptingOperatorMetricGroup metaspaceMetrics = new InterceptingOperatorMetricGroup();
+		final InterceptingOperatorMetricGroup parentMetrics = new InterceptingOperatorMetricGroup() {
+			@Override
+			public MetricGroup addGroup(String name) {
+				return metaspaceMetrics;
```

Comment:
Could we not return `this`?

Reply:
Good idea. I addressed it in the code, squashed all the changes into two commits and rebased the PR to the most recent master. Thanks for the review, @zentol .